src/components/users/add-member-dialog.tsx

```diff
@@ -15,120 +18,279 @@ interface Props {
-          { strings.userManagementScreen.addMemberDialog.text }
-        </DialogContentText>
-        <Stack spacing={ 1 }>
-          <TextField
-            fullWidth
-            size="medium"
-            required={ true }
-            value={ member.email }
-            type="email"
-            name="email"
-            label={ strings.userManagementScreen.addMemberDialog.emailLabel }
-            onChange={ onTextFieldChange }
-          />
-          <TextField
-            fullWidth
-            size="medium"
-            required={ true }
-            value={ member.firstName }
-            name="firstName"
-            label={ strings.userManagementScreen.addMemberDialog.firstNameLabel }
-            onChange={ onTextFieldChange }
-          />
+    setFoundUsers(users.length <= 10 ? users : users.slice(10));
+    setLoading(false);
+  };
+
+  /**
+   * Clears search textfield
+   */
+  const handleSearchClear = () => setUserSearch("");
+
+  /**
+   * Event handler for create button click
+   */
+  const handleCreateClick = async () => {
+    if (!selectedUser) {
+      return;
+    }
+
+    if (!selectedUser.id) {
+      const createdUser = await createUser({
+        ...selectedUser,
+        username: selectedUser.displayName ?? selectedUser.email
+      });
+
+      if (!createdUser) {
+        return;
+      }
+
+      onCreate({
+        firstName: createdUser.firstName,
+        lastName: createdUser.lastName,
+        email: createdUser.email,
+        role: MetaformMemberRole.Manager
+      });
+    } else {
+      onCreate({
+        firstName: selectedUser.firstName,
+        lastName: selectedUser.lastName,
+        email: selectedUser.email,
+        role: MetaformMemberRole.Manager
+      });
+    }
+
+    handleCancelClick();
+    setLoading(false);
+  };
+
+  /**
+   * Renders correct icon for User selection dialog
+   * 
+   * @param user User
+   */
+  const renderSelectOptionIcon = (user: User) => {
+    if (!user.id) {
+      return <CreditCardIcon/>;
+    }
+    const userIsLinkedToCard = user.federatedIdentities?.some(federatedIdentity => federatedIdentity.source === UserFederationSource.Card);
+    if (userIsLinkedToCard) {
+      return <LinkIcon/>;
+    }
+  };
+
+  /**
+   * Renders select box items
+   */
+  const renderSelectableUsers = (user: User) => (
+    <MenuItem key={ user.id ?? user.displayName } component={ ListItem } value={ user.id ?? user.displayName }>
+      <ListItemIcon>
+        { renderSelectOptionIcon(user) }
+      </ListItemIcon>
+      { user.displayName }
+    </MenuItem>
+  );
+
+  const valid = selectedUser?.firstName && selectedUser?.lastName && EmailValidator.validate(selectedUser?.email);
+
+  /**
+   * Renders dialog content
+   */
+  const renderDialogContent = () => (
+    <Stack spacing={ 1 }>
+      <Stack spacing={ 1 } direction="row" alignItems="center">
+        <Tooltip title="TOOLTIPPI">
```

Comment:
Maybe change this to "notYetImplemented"

Reply:
Actually that should've been removed all along. That was the tooltip we were testing yesterday.